packages/core/core/src/services/document-service/transform/__tests__/id-transform-i18n.test.ts

```diff
@@ -0,0 +1,199 @@
+      { id: 'category-9-en-draft', documentId: 'category-9', locale: 'en', publishedAt: null },
+      { id: 'category-10-en-draft', documentId: 'category-10', locale: 'en', publishedAt: null },
+    ]);
+
+    findProducts.mockReturnValue([
+      { id: 'product-1-draft', documentId: 'product-1', locale: null, publishedAt: null },
+      { id: 'product-1-published', documentId: 'product-1', locale: null, publishedAt: new Date() },
+      { id: 'product-2-draft', documentId: 'product-2', locale: null, publishedAt: null },
+      { id: 'product-3-draft', documentId: 'product-3', locale: null, publishedAt: null },
+    ]);
+  });
+
+  describe('Non I18n (products) -> I18n (categories)', () => {
+    it('Connect to locales of the same category document', async () => {
+      //
+      const { data } = await transformParamsDocumentId(
+        PRODUCT_UID,
+        {
+          data: {
+            name: 'test',
+            categories: [
+              { id: 'category-1', locale: 'en' },
+              { id: 'category-1', locale: 'fr' },
+              { id: 'category-2', locale: 'en' },
+            ],
+            category: { id: 'category-4', locale: 'en' },
+            relatedProducts: [{ id: 'product-1' }, { id: 'product-2', locale: null }],
+          },
+        },
+        { locale: 'en', isDraft: true }
+      );
+
+      expect(data).toMatchObject({
+        name: 'test',
+        categories: [
+          { id: 'category-1-en-draft' },
+          { id: 'category-1-fr-draft' },
+          { id: 'category-2-en-draft' },
+        ],
+        category: { id: 'category-4-en-draft' },
+        relatedProducts: [{ id: 'product-1-draft' }, { id: 'product-2-draft' }],
+      });
+    });
+
+    it('Connect to the default locale if not provided', async () => {
+      // Should connect to the default locale if not provided in the relation
+      const { data } = await transformParamsDocumentId(
+        PRODUCT_UID,
+        {
+          data: {
+            name: 'test',
+            categories: [{ id: 'category-1' }],
+          },
+        },
+        { isDraft: true }
+      );
+
+      expect(data).toMatchObject({
+        name: 'test',
+        categories: [{ id: 'category-1-en-draft' }],
+      });
+    });
+
+    it('Connect and reorder', async () => {
+      // Should connect and reorder the relations,
+      const { data } = await transformParamsDocumentId(
+        PRODUCT_UID,
+        {
+          data: {
+            name: 'test',
+            categories: {
+              connect: [
+                {
+                  id: 'category-1',
+                  locale: 'fr',
+                  position: { before: 'category-2', locale: 'en' },
+                },
+                { id: 'category-2', locale: 'en', position: { after: 'category-1', locale: 'fr' } },
+              ],
+            },
+          },
+        },
+        { isDraft: true }
```

Comment:
There is this special case when connecting `Non I18n CT`-> `I18n CT` 

Where it might be necessary to specify the locale both in the connected relation, but also in it's positional arguments

cc @alexandrebodin @joshuaellis

Reply:
because the ids aren't unique to the relation you're connecting? 🤔

Reply:
exactly, this would be an example:
![image](https://github.com/strapi/strapi/assets/20578351/c510a4e6-d65f-4d06-81a1-626bcb984b47)

From the category side, you should be able to connect both the english and spanish version, but both share the same document id. If you need to refernece them , you need to provide the doc id + locale